Author: Federico Ceratto <federico@debian.org>
Description: remove Sign In button from main HTML page

diff -Naurp netdata.orig/web/gui/main.js netdata/web/gui/main.js
--- netdata.orig/web/gui/main.js
+++ netdata/web/gui/main.js
@@ -4758,9 +4758,7 @@ function renderAccountUI() {
         container.setAttribute("data-original-title", "sign in");
         container.setAttribute("data-placement", "bottom");
         container.innerHTML = (
-            `<a href="#" class="btn" onclick="signInDidClick(event); return false">
-                <i class="fas fa-sign-in-alt"></i>&nbsp;<span class="hidden-sm hidden-md">Sign In</span>
-            </a>`
+            ``
         )
     }
 }
