From: Lennart Weller <lhw@ring0.de>
Date: Tue, 24 Jan 2017 16:48:02 +0100
Subject: linked js/css/fonts removed from make

---
 web/Makefile.am | 21 ---------------------
 1 file changed, 21 deletions(-)

diff --git a/web/Makefile.am b/web/Makefile.am
index 396cf0b..d94d87e 100644
--- a/web/Makefile.am
+++ b/web/Makefile.am
@@ -25,15 +25,6 @@ dist_web_DATA = \
 	version.txt \
 	$(NULL)
 
-webolddir=$(webdir)/old
-dist_webold_DATA = \
-	old/datasource.html \
-	old/index.html \
-	old/index.js \
-	old/netdata.js \
-	old/theme.css \
-	$(NULL)
-
 weblibdir=$(webdir)/lib
 dist_weblib_DATA = \
 	lib/bootstrap-3.3.7.min.js \
@@ -62,23 +53,11 @@ dist_webcss_DATA = \
 	css/bootstrap-theme-3.3.7.min.css \
 	css/bootstrap-slate-flat-3.3.7.css \
 	css/bootstrap-toggle-2.2.2.min.css \
-	css/font-awesome.min.css \
 	css/c3-0.4.11.min.css \
 	$(NULL)
 
 webfontsdir=$(webdir)/fonts
 dist_webfonts_DATA = \
-	fonts/glyphicons-halflings-regular.eot \
-	fonts/glyphicons-halflings-regular.svg \
-	fonts/glyphicons-halflings-regular.ttf \
-	fonts/glyphicons-halflings-regular.woff \
-	fonts/glyphicons-halflings-regular.woff2 \
-	fonts/FontAwesome.otf \
-	fonts/fontawesome-webfont.eot \
-	fonts/fontawesome-webfont.svg \
-	fonts/fontawesome-webfont.ttf \
-	fonts/fontawesome-webfont.woff \
-	fonts/fontawesome-webfont.woff2 \
 	$(NULL)
 
 webimagesdir=$(webdir)/images
