From: Lennart Weller <lhw@ring0.de>
Date: Wed, 25 May 2016 13:15:40 +0200
Subject: linked js/css/fonts removed from make

---
 web/Makefile.am | 27 ---------------------------
 1 file changed, 27 deletions(-)

--- a/web/Makefile.am
+++ b/web/Makefile.am
@@ -23,27 +23,14 @@
 	version.txt \
 	$(NULL)
 
-webolddir=$(webdir)/old
-dist_webold_DATA = \
-	old/datasource.html \
-	old/index.html \
-	old/index.js \
-	old/netdata.js \
-	old/theme.css \
-	$(NULL)
-
 weblibdir=$(webdir)/lib
 dist_weblib_DATA = \
-	lib/bootstrap-3.3.7.min.js \
 	lib/bootstrap-table-1.11.0.min.js \
 	lib/bootstrap-table-export-1.11.0.min.js \
 	lib/bootstrap-toggle-2.2.2.min.js \
-	lib/c3-0.4.11.min.js \
-	lib/d3-3.5.17.min.js \
 	lib/dygraph-combined-dd74404.js \
 	lib/dygraph-smooth-plotter-dd74404.js \
 	lib/gauge-d5260c3.min.js \
-	lib/jquery-2.2.4.min.js \
 	lib/jquery.easypiechart-97b5824.min.js \
 	lib/jquery.nanoscroller-0.8.7.min.js \
 	lib/jquery.peity-3.2.0.min.js \
@@ -56,27 +43,13 @@
 webcssdir=$(webdir)/css
 dist_webcss_DATA = \
 	css/morris-0.5.1.css \
-	css/bootstrap-3.3.7.min.css \
-	css/bootstrap-theme-3.3.7.min.css \
 	css/bootstrap.slate.min.css \
 	css/bootstrap-toggle-2.2.2.min.css \
-	css/font-awesome.min.css \
 	css/c3-0.4.11.min.css \
 	$(NULL)
 
 webfontsdir=$(webdir)/fonts
 dist_webfonts_DATA = \
-	fonts/glyphicons-halflings-regular.eot \
-	fonts/glyphicons-halflings-regular.svg \
-	fonts/glyphicons-halflings-regular.ttf \
-	fonts/glyphicons-halflings-regular.woff \
-	fonts/glyphicons-halflings-regular.woff2 \
-	fonts/FontAwesome.otf \
-	fonts/fontawesome-webfont.eot \
-	fonts/fontawesome-webfont.svg \
-	fonts/fontawesome-webfont.ttf \
-	fonts/fontawesome-webfont.woff \
-	fonts/fontawesome-webfont.woff2 \
 	$(NULL)
 
 webimagesdir=$(webdir)/images
