From: Lennart Weller <lhw@ring0.de>
Date: Wed, 4 Apr 2018 14:25:03 +0200
Subject: remove file serve restrictions for symlinks

---
 src/web_client.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/web_client.c b/src/web_client.c
index 477fb3d..38ebf47 100644
--- a/src/web_client.c
+++ b/src/web_client.c
@@ -362,7 +362,7 @@ int mysendfile(struct web_client *w, char *filename) {
     int done = 0;
     while(!done) {
         // check if the file exists
-        if (lstat(webfilename, &statbuf) != 0) {
+        if (stat(webfilename, &statbuf) != 0) {
             debug(D_WEB_CLIENT_ACCESS, "%llu: File '%s' is not found.", w->id, webfilename);
             w->response.data->contenttype = CT_TEXT_HTML;
             buffer_strcat(w->response.data, "File does not exist, or is not accessible: ");
