From: Lennart Weller <lhw@ring0.de>
Date: Wed, 14 Nov 2018 14:24:17 +0100
Subject: remove file serve restrictions for symlinks

---
 web/server/web_client.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/web/server/web_client.c b/web/server/web_client.c
index 282cfcd..eed6b59 100644
--- a/web/server/web_client.c
+++ b/web/server/web_client.c
@@ -372,7 +372,7 @@ int mysendfile(struct web_client *w, char *filename) {
     int done = 0;
     while(!done) {
         // check if the file exists
-        if (lstat(webfilename, &statbuf) != 0) {
+        if (stat(webfilename, &statbuf) != 0) {
             debug(D_WEB_CLIENT_ACCESS, "%llu: File '%s' is not found.", w->id, webfilename);
             w->response.data->contenttype = CT_TEXT_HTML;
             buffer_strcat(w->response.data, "File does not exist, or is not accessible: ");
